org-ql.el

```diff
@@ -448,6 +452,29 @@ Returns cons (INHERITED-TAGS . LOCAL-TAGS)."
                  org-ql-tags-cache))
       (puthash position all-tags tags-cache))))
 
+(defun org-ql--expand-tag-groups (tags &optional exclude-groups)
+  (if (eq tags 'org-ql-nil)
```

Comment:
This might be nitpicky, but I really do want to keep org-ql's performance in mind when writing code that gets called in tight loops, which this basically is, so: Before merging this, I'd like to benchmark this as-is and compare it to checking for `org-ql-nil` in the calling function.  Function call overhead is sometimes an issue in Emacs, and sometimes I do searches that can return hundreds or thousands of entries, in which case it can be noticeable.

Reply:
You certainly have a point that it would generally result in unnecessary function calls and, as such, I agree with you that the check should be performed inside the callers body regardless. Addressed in the new commit.